src/std/crypto/pkey.ss

```diff
@@ -42,24 +153,73 @@
     bytes)
    (else #f)))
 
+(def (BIO-read-all bio buff (parents []))
+  (let ((bend (u8vector-length buff))
+	(readn (BIO_read bio buff -1)))
+    (if (= bend readn)
+      (BIO-read-all bio (make-u8vector bend) (cons buff parents))
+      (let ((len (apply + readn (map u8vector-length parents)))
+	    (cats (reverse (cons buff parents))))
+	(if (< len 0) (make-u8vector 0)
+	    (bytes-result (u8vector-concatenate cats) len))))))
```

Comment:
it makes sense to explicitly deallocate the BIO buffer with `foreign-release!` in an unwind-protect postlude (or try/finally)